Migrated to nnbd

bsutton opened the pull request:
The vin library has released an nnbd version so I've migrated validators to nndb (given that 2.12 is now released).

Can you please publish as a preview.

Thanks.

bsutton committed:
Migrated to nnbd

bsutton commented:
I've published validators2 to pub.dev as a nnbd implementation until the authors have time to publish one of the two outstanding PRs for nnbd.

bsutton committed:
temporary release until main authors can publish nnbd version.

bsutton committed:
Released 3.0.0

anotherpit commented:
@bsutton It seems like after your recent commits this PR couldn't be merged into this repo anymore (as now it has conflicts, and also changes the library name to `validators2). I'd suggest creating a new PR from your commit https://github.com/dart-league/validators/pull/24/commits/091130e203e561b1ad8afbbce8915b0009a8f718. And thanks for your job, I really need a null safe version of this library.

bsutton committed:
Merge remote-tracking branch 'upstream/master' and fixed package name.

bsutton commented:
resolved the merge conflicts and fixed the package name.

anotherpit commented:
@bsutton A new version got published several days ago, that supports null safety.

bsutton committed:
ignored .failed_tracker

bsutton committed:
Modernised the api to confirm to post 2.12 best practices.
There are a number of small breaking changes, the main theme is changing optional args to named args with a default value.

bsutton committed:
Released 4.0.0.

bsutton committed:
Released 4.0.0.

bsutton committed:
resolved inconsitency in .gitignore add .pubignore to remove travis.yml from published files.

bsutton committed:
Released 5.0.0.

bsutton committed:
upgraded packges, fixed ilnts.

bsutton committed:
Released 5.1.0.
